frontend/src/general/SnackbarMessage.jsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Snackbar, SnackbarContent, IconButton, Box } from "@material-ui/core";
+import { CheckCircle, Error, Close } from "@material-ui/icons";
+import { withRouter } from "react-router";
+
+let openSnackbarFn;
+
+class SnackbarMessage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      isSuccess: false,
+      status: "",
+      message: "",
+      pathname: ""
+    };
+    this.setOpen = this.setOpen.bind(this);
+    this.setIsSuccess = this.setIsSuccess.bind(this);
+    this.setStatus = this.setStatus.bind(this);
+    this.setMessage = this.setMessage.bind(this);
+    this.setPathname = this.setPathname.bind(this);
+    this.addSnackbarProperties = this.addSnackbarProperties.bind(this);
+    this.openSnackbar = this.openSnackbar.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.refresh = this.refresh.bind(this);
+  }
+
+  componentDidMount() {
+    openSnackbarFn = this.openSnackbar;
+  }
+
+  handleClose(event, reason) {
+    if (reason === "clickaway") {
+      return;
+    }
+    this.setOpen(false);
+  }
+
+  setOpen(open) {
+    this.setState({
+      open
+    });
+  }
+
+  setStatus(status) {
+    this.setState({
+      status
+    });
+  }
+
+  setIsSuccess(isSuccess) {
+    this.setState({
+      isSuccess
+    });
+  }
+
+  setMessage(message) {
+    this.setState({
+      message
+    });
+  }
+
+  setPathname(pathname) {
+    this.setState({
+      pathname
+    });
+    return pathname;
+  }
+
+  addSnackbarProperties(stat, msg) {
+    this.setStatus(stat);
+    if (stat !== "error") {
+      this.setIsSuccess(true);
+    } else {
+      this.setIsSuccess(false);
+    }
+    this.setMessage(msg);
+  }
+
+  openSnackbar(stat, msg, path) {
+    const currentPathname = this.setPathname(path);
+    if (currentPathname !== "") {
```

Comment:
Why make `this.setPathname` return the pathname? I'd suggest instead doing
```
this.setPathname(path);
if (path !== "") {
```

Reply:
This is because if we do this the pathname will be evaluated at the same time as the set is being made and thus will mess up with the if statement evaluation.

Reply:
After some testing with @felixlapierre we discovered that this can no longer be replicated, so the above comment is no longer valid.